sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -536,6 +538,134 @@ class CollationStringExpressionsSuite
+      StringTrimTestCase("UNICODE_CI", "BTRIM", "asd", false, null, "asd"),
+      StringTrimTestCase("UNICODE_CI", "BTRIM", "  asd  ", false, null, "asd"),
+      StringTrimTestCase("UNICODE_CI", "LTRIM", "asd", false, null, "asd"),
+      StringTrimTestCase("UNICODE_CI", "LTRIM", "  asd  ", false, null, "asd  "),
+      StringTrimTestCase("UNICODE_CI", "RTRIM", "asd", false, null, "asd"),
+      StringTrimTestCase("UNICODE_CI", "RTRIM", "  asd  ", false, null, "  asd"),
+
+      // With null trimString param.
+      StringTrimTestCase("UNICODE_CI", "TRIM", "asd", true, null, null),
+      StringTrimTestCase("UNICODE_CI", "TRIM", "  asd  ", true, null, null),
+      StringTrimTestCase("UNICODE_CI", "BTRIM", "asd", true, null, null),
+      StringTrimTestCase("UNICODE_CI", "BTRIM", "  asd  ", true, null, null),
+      StringTrimTestCase("UNICODE_CI", "LTRIM", "asd", true, null, null),
+      StringTrimTestCase("UNICODE_CI", "LTRIM", "  asd  ", true, null, null),
+      StringTrimTestCase("UNICODE_CI", "RTRIM", "asd", true, null, null),
+      StringTrimTestCase("UNICODE_CI", "RTRIM", "  asd  ", true, null, null),
+
+      // With " " trimString param.
+      StringTrimTestCase("UNICODE_CI", "TRIM", "asd", true, " ", "asd"),
+      StringTrimTestCase("UNICODE_CI", "TRIM", "  asd  ", true, " ", "asd"),
+      StringTrimTestCase("UNICODE_CI", "BTRIM", "asd", true, " ", "asd"),
+      StringTrimTestCase("UNICODE_CI", "BTRIM", "  asd  ", true, " ", "asd"),
+      StringTrimTestCase("UNICODE_CI", "LTRIM", "asd", true, " ", "asd"),
+      StringTrimTestCase("UNICODE_CI", "LTRIM", "  asd  ", true, " ", "asd  "),
+      StringTrimTestCase("UNICODE_CI", "RTRIM", "asd", true, " ", "asd"),
+      StringTrimTestCase("UNICODE_CI", "RTRIM", "  asd  ", true, " ", "  asd")
+
+      // Other more complex cases can be found in unit tests in CollationSupportSuite.java.
+    )
+
+    // scalastyle:off
+
+    testCases.foreach(testCase => {
+      var df: DataFrame = null
+
+      if (testCase.trimFunc.equalsIgnoreCase("BTRIM")) {
+        // BTRIM has arguments in (srcStr, trimStr) order
+        df = sql(s"SELECT ${testCase.trimFunc}(" +
+          s"COLLATE('${testCase.sourceString}', '${testCase.collation}')" +
+          (if (!testCase.hasTrimString) "" else if (testCase.trimString == null) ", null" else s", COLLATE('${testCase.trimString}', '${testCase.collation}')") +
+          ")")
+      }
+      else {
+        // While other functions have arguments in (trimStr, srcStr) order
+        df = sql(s"SELECT ${testCase.trimFunc}(" +
+          (if (!testCase.hasTrimString) "" else if (testCase.trimString == null) "null, " else s"COLLATE('${testCase.trimString}', '${testCase.collation}'), ") +
+          s"COLLATE('${testCase.sourceString}', '${testCase.collation}')" +
+          ")")
+      }
+
+      checkAnswer(df = df, expectedAnswer = Row(testCase.expectedResultString))
+    })
+
```

Comment:
also, collation mismatch test is missing here

Reply:
Added.